### PR TITLE
resolved-ts: add regression test for representative large txn unlock

### PR DESCRIPTION
**Pull request by younes-io**
The resolver keeps only one representative key for a large transaction, and `untrack_lock` removes the whole large-txn entry when that representative key unlocks. If that key unlocks before other keys from the same transaction, the resolver can forget a still-live large transaction and advance `resolved_ts` too far. I could not find a documented invariant stating the representative key must unlock last, and a small regression test reproduces the issue on this PR 

Issue Number: ref https://github.com/tikv/tikv/issues/19495

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented premature advancement of resolved timestamps for large transactions when only a representative key is unlocked.
* **Tests**
  * Added coverage for edge-case large-transaction resolved-timestamp and cleanup behavior.

### CODE REVIEW

File: components/resolved_ts/src/resolver.rs [hunk 1]
```diff
@@ -953,4 +953,46 @@ mod tests {
...
+
+        assert_eq!(
+            resolver.resolve(2.into(), None, TsSource::PdTso),
+            1.into()
+        );
+
+        // The remaining key can then be untracked without affecting the invariant
+        // we are checking above.
+        resolver.untrack_lock(&trailing_key, None);
+    }
```
[the remark this answers is not in the document]
Comment (younes-io) [reply components/resolved_ts/src/resolver.rs:1032]: this is from old comment; The test is no longer marked #[ignore], now runs as a normal unit test, and passes
